base/loading.jl

```diff
@@ -2940,6 +2964,17 @@ end
             """
             return true
         end
+        if CacheFlags(flags).is_mandatory != 0x0 && !allow_new_mandatory
+            # Check whether this is in the current list of mandatory packages
+            for (mandatory_modkey, mandatory_build_id) in _mandatory_dependencies
+                if mandatory_modkey == modkey && mandatory_build_id == build_id
+                    @goto mandatory_ok
+                end
+            end
+            @debug "Rejecting cache file $cachefile for $modkey since it is a mandatory package that we have not loaded"
+            return true
+            @label mandatory_ok
+        end
```

Comment:
```suggestion
            # Check whether this is in the current list of mandatory packages
            found = false
            for (mandatory_modkey, mandatory_build_id) in _mandatory_dependencies
                if mandatory_modkey == modkey && mandatory_build_id == build_id
                    found = true
                    break
                end
            end
            if !found
                @debug "Rejecting cache file $cachefile for $modkey since it is a mandatory package that we have not loaded"
                return true
            end
        end
```

Since no other piece of this code uses gotos